OData/src/System.Web.OData/OData/Query/Expressions/FilterBinder.cs

```diff
@@ -233,6 +237,57 @@ private Expression Bind(QueryNode node)
             }
         }
 
+        private Expression BindOpenPropertyAccessQueryNode(SingleValueOpenPropertyAccessNode openNode)
+        {
+            var prop = GetOpenTypeProperty(openNode);
+            var propertyAccessExpression = BindPropertyAccessExpression(openNode, prop);
+
+            var dynamicDictIsNotNull = Expression.NotEqual(propertyAccessExpression, Expression.Constant(null));
+            var containsKeyExpression = Expression.Call(propertyAccessExpression, propertyAccessExpression.Type.GetMethod("ContainsKey"), Expression.Constant(openNode.Name));
+            var dynamicDictExitsAndIsNotNull = Expression.AndAlso(dynamicDictIsNotNull, containsKeyExpression);
+            var nullExpression = Expression.Constant(null);
+            var readDictionaryIndexerExpression = Expression.Property(propertyAccessExpression, _dictionaryStringObjectIndexerName, Expression.Constant(openNode.Name));
+            return Expression.Condition(
+                dynamicDictExitsAndIsNotNull,
+                readDictionaryIndexerExpression,
+                nullExpression
+                );
+        }
+
+        private Expression BindPropertyAccessExpression(SingleValueOpenPropertyAccessNode openNode, PropertyInfo prop)
+        {
+            var source = Bind(openNode.Source);
+            Expression propertyAccessExpression;
+            if (_querySettings.HandleNullPropagation == HandleNullPropagationOption.True &&
+                IsNullable(source.Type) && source != _lambdaParameters[ODataItParameterName])
+                propertyAccessExpression = Expression.Property(RemoveInnerNullPropagation(source), prop.Name);
+            else
+            {
+                propertyAccessExpression = Expression.Property(source, prop.Name);
+            }
+            return propertyAccessExpression;
+        }
+
+        private PropertyInfo GetOpenTypeProperty(SingleValueOpenPropertyAccessNode openNode)
```

Comment:
make change it as "GetDynamicPropertyDictionary" or "GetDynamicPropertyContainer()"
or make some comment for this function